OtripleS.Portal.Web.Tests.Unit/Services/Courses/CourseServiceTests.cs

```diff
@@ -0,0 +1,109 @@
+
+        public CourseServiceTests()
+        {
+            this.apiBrokerMock = new Mock<IApiBroker>();
+            this.loggingBrokerMock = new Mock<ILoggingBroker>();
+
+            this.courseService = new CourseService(
+                apiBroker: this.apiBrokerMock.Object,
+                loggingBroker: this.loggingBrokerMock.Object);
+        }
+
+        public static TheoryData CriticalApiException()
+        {
+            string exceptionMessage = GetRandomString();
+            var responseMessage = new HttpResponseMessage();
+
+            var httpRequestException =
+                new HttpRequestException();
+
+            var httpResponseUrlNotFoundException =
+                new HttpResponseUrlNotFoundException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            var httpResponseUnAuthorizedException =
+                new HttpResponseUnauthorizedException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            return new TheoryData<Exception>
+            {
+                httpRequestException,
+                httpResponseUrlNotFoundException,
+                httpResponseUnAuthorizedException
+            };
+        }
+
+        public static TheoryData DependencyApiException()
+        {
+            string exceptionMessage = GetRandomString();
+            var responseMessage = new HttpResponseMessage();
+
+            var httpResponseException =
+                new HttpResponseException(
+                    httpResponseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            var httpResponseInternalServerErrorException =
+                new HttpResponseInternalServerErrorException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            return new TheoryData<Exception>
+            {
+                httpResponseException,
+                httpResponseInternalServerErrorException
+            };
+        }
+
+        private static List<Course> CreateRandomCourses() =>
+            CreateCourseFiller().Create(count: GetRandomNumber()).ToList();
+
+        private static Expression<Func<Exception, bool>> SameExceptionAs(
+            Exception expectedException)
+        {
+            return actualException => actualException.Message == expectedException.Message
+                && actualException.InnerException.Message == expectedException.InnerException.Message;
+        }
+
+        private static string GetRandomString() => new MnemonicString().GetValue();
+        private static int GetRandomNumber() => new IntRange(min: 2, max: 10).GetValue();
+
+        private static Filler<Course> CreateCourseFiller()
+        {
+            var filler = new Filler<Course>();
+
+            filler.Setup()
+                .OnType<DateTimeOffset>().Use(DateTimeOffset.UtcNow);
```

Comment:
We should be using random time here. Example - https://github.com/hassanhabib/OtripleS.Portal/blob/master/OtripleS.Portal.Web.Tests.Unit/Services/Teachers/TeacherServiceTests.cs

Reply:
@ShriHumrudha thank you for catching this. resoled